src/Utilities/ErrorHandling/CMakeLists.txt

```diff
@@ -32,3 +32,9 @@ target_link_libraries(
   PUBLIC
   SystemUtilities
   )
+
+# For `backtrace` functionality
+if (APPLE)
+  target_compile_options(${LIBRARY} INTERFACE "-dynamic")
+  target_link_options(${LIBRARY} INTERFACE "-rdynamic")
```

Comment:
I believe the deprecation applies only to executables. Note that the deprecation was applied in CMake v3.4, and since we require v3.12 the `NEW` behavior is automatically applied.

Reply:
"Unlike most policies, CMake version 3.22.2 does not warn by default when this policy is not set and simply uses OLD behavior."  Sounds like the NEW behavior needs to be specifically requested, unless I'm misunderstanding how policies work (which I might be).

I think you're right about executables vs. libraries, though, so this does look correct.

Reply:
I think all policies introduced before v3.12 are automatically set to `NEW`, if I understand the docs right: https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html

Reply:
Yes, that does look like what it says.  That call counts as "setting" everything, so the line I quoted doesn't apply.